src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/trial_of_the_champion.cpp

```diff
@@ -360,28 +546,269 @@ class npc_announcer_toc5 : public CreatureScript
             }
         }
 
-        void AggroAllPlayers(Creature* temp)
+        void StartRpEvent()
         {
-            Map::PlayerList const &PlList = me->GetMap()->GetPlayers();
+            me->RemoveFlag(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_GOSSIP);
+            me->AI()->Talk(SAY_INTRO_1);
+            NextStep(7000, false, 1);
+        }
 
-            if (PlList.isEmpty())
-                return;
+        void IntroduceChampion()
+        {
+            uint32 TeamInInstance = instance->GetData(DATA_PLAYERS_TEAM);
 
-            for (Map::PlayerList::const_iterator i = PlList.begin(); i != PlList.end(); ++i)
+            if (GrandChampionList[uiBoss-1] == 0)
+            {
+                if (TeamInInstance == ALLIANCE)
```

Comment:
what about to create method for it, with textid and npc entry as arguments? you can safe 300 lines